src/utils/MSFraggerAdapter.cpp

```diff
@@ -0,0 +1,859 @@
+
+    // TOPP tolerance
+    registerTOPPSubsection_("tolerance", "Search Tolerances");
+
+    // Precursor mass tolerance and unit
+    _registerNonNegativeDouble(TOPPMSFraggerAdapter::param_precursor_mass_tolerance, "<precursor_mass_tolerance>", 20.0, "Precursor mass tolerance", false, false);
+    registerStringOption_(TOPPMSFraggerAdapter::param_precursor_mass_unit, "<precursor_mass_unit>", "ppm", "Unit of precursor mass tolerance", false, false);
+    setValidStrings_(TOPPMSFraggerAdapter::param_precursor_mass_unit, validUnits);
+
+    // Precursor true tolerance
+    _registerNonNegativeDouble(TOPPMSFraggerAdapter::param_precursor_true_tolerance, "<precursor_true_tolerance>", 0.0, "Precursor true tolerance. It is strongly recommended to set this feature!", false, false);
+    registerStringOption_(TOPPMSFraggerAdapter::param_precursor_true_unit, "<precursor_true_unit>", "ppm", "Unit of precursor true tolerance", false, false);
+    setValidStrings_(TOPPMSFraggerAdapter::param_precursor_true_unit, validUnits);
+
+    // Fragment mass tolerance
+    _registerNonNegativeDouble(TOPPMSFraggerAdapter::param_fragment_mass_tolerance, "<fragment_mass_tolerance>", 20.0, "Fragment mass tolerance", false, false);
+    registerStringOption_(TOPPMSFraggerAdapter::param_fragment_mass_unit, "<fragment_mass_unit>", "ppm", "Fragment mass unit", false, false);
+    setValidStrings_(TOPPMSFraggerAdapter::param_fragment_mass_unit, validUnits);
+
+    // Isotope error
+    registerStringOption_(TOPPMSFraggerAdapter::param_isotope_error, "<isotope_error>", "0", "Isotope correction for MS/MS events triggered on isotopic peaks. Should be set to 0 for open search or 0/1/2 for correction of narrow window searches", false, false);
+    setValidStrings_(TOPPMSFraggerAdapter::param_isotope_error, isotope_error_and_enzyme_termini);
+
+    // TOPP digest
+    registerTOPPSubsection_("digest", "In-Silico Digestion Parameters");
+
+    // Enzyme
+    StringList enzyme_names;
+    EnzymesDB::getInstance()->getAllNames(enzyme_names);
+    registerStringOption_(TOPPMSFraggerAdapter::param_search_enzyme_name, "<search_enzyme_name>", "Trypsin", "Name of the enzyme to be written to the pepXML file", false, false);
+    setValidStrings_(TOPPMSFraggerAdapter::param_search_enzyme_name, enzyme_names);
+
+    // Cut after
+    registerStringOption_(TOPPMSFraggerAdapter::param_search_enzyme_cutafter, "<search_enzyme_cutafter>", "KR", "Residues after which the enzyme cuts", false , false);
+
+    // No cut before
+    registerStringOption_(TOPPMSFraggerAdapter::param_search_enzyme_nocutbefore, "<search_enzyme_nocutbefore>", "P", "Residues that the enzyme will not cut before", false, false);
+
+    // Number of enzyme termini
+    registerStringOption_(TOPPMSFraggerAdapter::param_num_enzyme_termini, "<num_enzyme_termini>", "2", "Number of enzyme termini (0, 1, or 2 for non-enzymatic, semi-enzymatic, fully-enzymatic)", false, false);
```

Comment:
In the Comet and MSGFPlus valid strings are used and in Comet mapped to integers later:
```
setValidStrings_("num_enzyme_termini", ListUtils::create<String>("semi,fully,C-term unspecific,N-term unspecific"));
    map<string,int> num_enzyme_termini;
    num_enzyme_termini["semi"] = 1;
    num_enzyme_termini["fully"] = 2;

```

Reply:
Is addressed.